courses/serializers.py

```diff
@@ -39,15 +40,52 @@ def to_internal_value(self, value):
         return obj
 
 
+class StringyManyToManyField(serializers.Field):
+    """
+    Serializes out as a list of strings on the way out, but reifies into
+    real models on the way in.
+    """
+    def __init__(self, *args, **kwargs):
+        self.model = kwargs.pop('model', None)
+        self.prop = kwargs.pop('lookup', None)
+
+        if not self.model:
+            raise ImproperlyConfigured("You must provide a model to use for reifying")
+
+        if not self.prop:
+            raise ImproperlyConfigured("You must provide a field lookup to "
+                                       "convert string value into model instance.")
+
+        super(StringyManyToManyField, self).__init__(*args, **kwargs)
+
+    def to_representation(self, obj):
+        """Python object to string"""
+        return (str(i) for i in obj.all())
+
+    def to_internal_value(self, value):
+        """Incoming value to python value"""
+        if not isinstance(value, list):
+            value = [value]
```

Comment:
It looks like that's not strictly true, I can pass in a `{}` via REST and it goes all the way to `to_internal_value`. Here's the test I wrote to investigate this:

    def test_invalid_type(self):
        from courses.models import EdxAuthor
        author = EdxAuthor.objects.create()
        resp = self.client.post("/api/v1/coursexs/", {
            "title": "title",
            "author_name": "author",
            "overview": "overview",
            "description": "description",
            "video_url": "http://x.com",
            "edx_instance": "http://x.com",
            "price_per_seat_cents": 3,
            "instructors": [author.edx_uid]
        })
        course_dict = json.loads(resp.content.decode('utf-8'))
        self.assertEqual(resp.status_code, 201)

        url = "/api/v1/coursexs/{uuid}/".format(uuid=course_dict['uuid'])
        resp = self.client.patch(url, json.dumps({
            "instructors": {}
        }), content_type='application/json')
        self.assertEqual(resp.status_code, 200)

And the stack trace:

    Traceback (most recent call last):
      File "/home/george/Projects/ccxcon/courses/tests/test_course.py", line 40, in test_hash
        }), content_type='application/json')
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/test/client.py", line 558, in patch
        secure=secure, **extra)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/test/client.py", line 345, in patch
        secure=secure, **extra)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
        return self.request(**r)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/test/client.py", line 466, in request
        six.reraise(*exc_info)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
        response = wrapped_callback(request, *callback_args, **callback_kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
        return view_func(*args, **kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/viewsets.py", line 87, in view
        return self.dispatch(request, *args, **kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/views.py", line 466, in dispatch
        response = self.handle_exception(exc)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/views.py", line 463, in dispatch
        response = handler(request, *args, **kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/mixins.py", line 78, in partial_update
        return self.update(request, *args, **kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/mixins.py", line 69, in update
        serializer.is_valid(raise_exception=True)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 213, in is_valid
        self._validated_data = self.run_validation(self.initial_data)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 407, in run_validation
        value = self.to_internal_value(data)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/serializers.py", line 437, in to_internal_value
        validated_value = field.run_validation(primitive_value)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/rest_framework/fields.py", line 482, in run_validation
        value = self.to_internal_value(data)
      File "/home/george/Projects/ccxcon/courses/serializers.py", line 72, in to_internal_value
        model, _ = self.model.objects.get_or_create(**{self.prop: v})
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
        return getattr(self.get_queryset(), name)(*args, **kwargs)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 405, in get_or_create
        return self.get(**lookup), False
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 328, in get
        num = len(clone)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 144, in __len__
        self._fetch_all()
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 965, in _fetch_all
        self._result_cache = list(self.iterator())
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/query.py", line 238, in iterator
        results = compiler.execute_sql()
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 840, in execute_sql
        cursor.execute(sql, params)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
        return self.cursor.execute(sql, params)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/utils.py", line 97, in __exit__
        six.reraise(dj_exc_type, dj_exc_value, traceback)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
        return self.cursor.execute(sql, params)
      File "/home/george/Projects/ccxcon_venv/local/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py", line 318, in execute
        return Database.Cursor.execute(self, query, params)
    InterfaceError: Error binding parameter 0 - probably unsupported type.